ChartIQDemo/ChartIQDemo/ViewController.swift

```diff
@@ -549,6 +549,12 @@ extension ViewController: ChartIQDelegate {
             chartIQView.setRefreshInterval(refreshInterval)
             chartIQView.setDataMethod(.pull)
             chartIQView.setSymbol(defaultSymbol)
+            
+            // if .push is used you'll probably want to make use of the chartIQView.isChartAvailable() method for your initial data push.
+            /*************************************************
+             * if(chartIQView.isChartAvailable() { pushData }
+             * else{ repeat check via recursion }
+            *************************************************/
```

Comment:
Are you trying to tell the developer not to push data until this function returns true?  Isn't this just pushing the dirty setTimeout back on them?

Why don't we just have the bridge handle the initial push data?  If chartAvailable is true, it is sent to loadChart() right away.  If it is false, it's stored in an array until setChartAvailable(true) is called for the first time.

Reply:
I'm also not understanding why this comment is in this function.  This function sets the dataMethod to .pull.  Why would you care about .push here?
Looking at this demo further, is it true that even if you were using .push, the initial data comes from the quotefeed?

Reply:
I guess I'm not really sure what you want to have the client do to fix the issue.

Reply:
I was just thinking about giving them the tools to check when a chart is available. If the comment is confusing I can take it out. I thought it might be clear as a use case but it doesn't seem to be the case. 

Handling the initialLoad upon setChartAvailable seems like a race condition just waiting to happen. We are running out of time but I'll try and mock something up.